chromium_src/chrome/browser/renderer_context_menu/render_view_context_menu.cc

```diff
@@ -406,103 +567,54 @@ void BraveRenderViewContextMenu::ExecuteAIChatCommand(int command) {
-      helper->SubmitSelectedText(
-          base::UTF16ToUTF8(params_.selection_text),
-          ai_chat::mojom::ActionType::CREATE_SOCIAL_MEDIA_COMMENT_LONG);
-      break;
-    case IDC_AI_CHAT_CONTEXT_IMPROVE:
-      p3a_action = ai_chat::ContextMenuAction::kImprove;
-      helper->SubmitSelectedText(base::UTF16ToUTF8(params_.selection_text),
-                                 ai_chat::mojom::ActionType::IMPROVE);
-      break;
-    case IDC_AI_CHAT_CONTEXT_ACADEMICIZE:
-      p3a_action = ai_chat::ContextMenuAction::kChangeTone;
-      helper->SubmitSelectedText(base::UTF16ToUTF8(params_.selection_text),
-                                 ai_chat::mojom::ActionType::ACADEMICIZE);
-      break;
-    case IDC_AI_CHAT_CONTEXT_PROFESSIONALIZE:
-      p3a_action = ai_chat::ContextMenuAction::kChangeTone;
-      helper->SubmitSelectedText(base::UTF16ToUTF8(params_.selection_text),
-                                 ai_chat::mojom::ActionType::PROFESSIONALIZE);
-      break;
-    case IDC_AI_CHAT_CONTEXT_PERSUASIVE_TONE:
-      p3a_action = ai_chat::ContextMenuAction::kChangeTone;
-      helper->SubmitSelectedText(base::UTF16ToUTF8(params_.selection_text),
-                                 ai_chat::mojom::ActionType::PERSUASIVE_TONE);
-      break;
-    case IDC_AI_CHAT_CONTEXT_CASUALIZE:
-      p3a_action = ai_chat::ContextMenuAction::kChangeTone;
-      helper->SubmitSelectedText(base::UTF16ToUTF8(params_.selection_text),
-                                 ai_chat::mojom::ActionType::CASUALIZE);
-      break;
-    case IDC_AI_CHAT_CONTEXT_FUNNY_TONE:
-      p3a_action = ai_chat::ContextMenuAction::kChangeTone;
-      helper->SubmitSelectedText(base::UTF16ToUTF8(params_.selection_text),
-                                 ai_chat::mojom::ActionType::FUNNY_TONE);
-      break;
-    case IDC_AI_CHAT_CONTEXT_SHORTEN:
-      p3a_action = ai_chat::ContextMenuAction::kChangeLength;
-      helper->SubmitSelectedText(base::UTF16ToUTF8(params_.selection_text),
-                                 ai_chat::mojom::ActionType::SHORTEN);
-      break;
-    case IDC_AI_CHAT_CONTEXT_EXPAND:
-      p3a_action = ai_chat::ContextMenuAction::kChangeLength;
-      helper->SubmitSelectedText(base::UTF16ToUTF8(params_.selection_text),
-                                 ai_chat::mojom::ActionType::EXPAND);
-      break;
-    default:
-      NOTREACHED_NORETURN();
-  }
-
-  if (p3a_action) {
-    g_brave_browser_process->process_misc_metrics()
-        ->ai_chat_metrics()
-        ->RecordContextMenuUsage(*p3a_action);
-  }
+  // To do rewrite in-place, the following conditions must be met:
+  // 1) Selected content is editable.
+  // 2) User has opted in to Leo.
+  // 3) SSE is enabled, it is required otherwise the UI update will be too slow.
+  // 4) The command is a rewrite command.
+  // 5) There's no in-progress in-place rewrite.
+  bool rewrite_in_place =
+      params_.is_editable &&
+      ai_chat::HasUserOptedIn(GetProfile()->GetPrefs()) &&
```

Comment:
Pls do a follow up issue to allow this but it should trigger some UI. Then tag Agustin for design-needed w/ label pls and thanks.

Reply:
https://github.com/brave/brave-browser/issues/37260